juliangruber: if it was working as expected, it would output 'success!' without throwing anything